docs/operate/get-started/setup-micro.md

```diff
@@ -63,8 +64,394 @@ To get started quickly with the pre-built `viam-micro-server` binary, follow the
+Upload the generated firmware to your ESP32:
+
+1. Use a data cable to connect your ESP32 board to a USB port on your development machine.
+
+1. Run the following command to flash the firmware to your ESP32:
+
+   ```sh { class="command-line" data-prompt="$"}
+   make flash-esp32-bin
+   ```
+
+   When prompted, select the serial port that your ESP32 is connected to through a data cable.
+
+   If the flash is successful, you will retain a serial connection to the board until you press `Ctrl-C`.
+   To manage this connection, consider running it within a dedicated terminal session, or under `tmux` or `screen`.
+   While the serial connection is live, you can also restart the currently flashed image with `Ctrl-R`.
+
+1. Navigate to your new machine's page on the [Viam app](https://app.viam.com).
+   If successful, the status indicator should turn green and show **Live**.
+
+After adding (or removing) a module or changing the version of a module, you must rerun the build and flash steps to rebuild the firmware and reflash the device.
+
+### Configure and test your machine
+
+You can now configure the models you included in your firmware and test them:
+
+1. Navigate to your machine's page in the [Viam app](https://app.viam.com).
+
+1. From the **CONFIGURE** tab, click **JSON** mode.
+   Micro-RDK components and services must be configured in JSON.
+
+1. Add your components and services to the machine configuration.
+   If you are using one of the example modules, find configuration details in the [example modules README](https://github.com/viamrobotics/micro-rdk/blob/main/examples/modular-drivers/README.md#example-modules).
+   For example, if you want to use a free heap sensor, your configuration could look like this:
+
+   ```json
+   {
+     "components": [
+       {
+         "name": "my-free-heap-sensor",
+         "api": "rdk:component:sensor",
+         "model": "free-heap",
+         "attributes": {}
+       }
+     ]
+   }
+   ```
+
+1. Click **Builder** mode and find the configuration card of the component you want to test.
+
+1. Click to open the **Test** section of the card and use the interface to test the component.
+
+## Build firmware with cloud build
+
+When developing and testing your firmware, it is easiest to build firmware locally.
+When you are ready to deploy your firmware to a fleet of microcontrollers, you can use the Viam cloud build service to simplify the process of building and hosting your firmware by eliminating the need to build locally and then upload the firmware image to a cloud storage bucket.
+
+To set up cloud build, follow these steps:
+
+1. Create a GitHub repository for any module you created and want to use in your firmware.
+   Upload the contents of the module directory to the repository.
+
+1. Update your firmware project's `Cargo.toml` file to use the URL of the GitHub repository, for example:
+
+   ```toml
+   [dependencies]
+   ...
+   my-module = { git = "https://github.com/username/my-module.git" }
```

Comment:
[note] I had wanted to use a concrete example module here, but I think it'll be better to do that after we move the example modules out of the micro-rdk repo, so this sounds fine for now

Feel free to resolve